voctocore/lib/commands.py

```diff
@@ -60,11 +87,38 @@ def set_composite_mode(self, composite_mode):
 		self.pipeline.vmix.setCompositeMode(mode)
 		return True
 
-	def set_stream_blank(self, src_name_or_id):
-		src_id = self.decodeBlankerSourceName(src_name_or_id)
-		self.pipeline.streamblanker.setBlankSource(src_id)
-		return True
+	def get_composite_mode(self):
+		try:
+			mode = self.pipeline.vmix.getCompositeMode()
+			name = str(mode).split('.')[-1]
+			return (True, name)
+		except Exception as e:
+			raise KeyError("composite-mode %s unknown" % mode)
+
+	def set_stream_status(self, *args):
+		try:
+			if args[0] == "live":
+				self.pipeline.streamblanker.setBlankSource(None)
+				return True
+			elif args [0] == "blank":
+				src_id = self.decodeBlankerSourceName(args[1])
+				self.pipeline.streamblanker.setBlankSource(src_id)
+				return True
+			else:
+				return (False, "invalid arguments")
+		except KeyError as e:
+			raise KeyError("invalid arguments")
+
+	def get_stream_status(self):
+		if self.pipeline.streamblanker.blankSource is None:
+			return (True, "live")
+
+		name = self.encodeBlankerSourceName(self.pipeline.streamblanker.blankSource)
+		return (True, "blank " + name)
+
+	def get_config(self):
+		dump = StringIO()
+		Config.write(dump)
+		dump.seek(0)
+		return (True, base64.b64encode(dump.read().encode()).decode())
```

Comment:
better build a dict here and use json as transport format: `{k: dict(v) for k, v in  dict(Config).items()}`
you can use `read_dict` on the gui to overlay the server config over the gui config.